include/common/array.h

```diff
@@ -41,4 +41,18 @@ wl_array_len(struct wl_array *array)
 		pos && (const char *)pos >= (const char *)(array)->data;               \
 		(pos)--)
 
+/**
+ * xwl_array_add() - add item to array and crash on allocation error
+ * @_arr: wl_array to add the item to
+ * @_val: the item to add to the array
+ * Note: asserts() when the memory allocation for the new item failed
+ */
+#define xwl_array_add(_arr, _val) ({                       \
```

Comment:
Now looks like ({}) GNU statement expression.

Are we sure we want that? I don’t personally mind them, but it does move us into a space some people resist.

Reply:
What do you suggest instead, a `do { .. } while (0)` loop?